k2/csrc/fsa_util.cc

```diff
@@ -34,4 +116,53 @@ void GetEnteringArcs(const Fsa &fsa, std::vector<int32_t> *arc_index,
   }
 }
 
+std::unique_ptr<Fsa> StringToFsa(const std::string &s) {
+  static constexpr const char *kDelim = " \t";
+
+  std::istringstream is(s);
+  std::string line;
+  std::vector<std::vector<Arc>> vec;  // index is state number
+  bool finished = false;  // when the final state is read, set it to true.
+  int32_t num_arcs = 0;
+  while (std::getline(is, line)) {
+    CHECK_EQ(finished, false);
+    std::vector<std::string> splits;
+    SplitStringToVector(line, kDelim, &splits);
+    if (splits.empty()) continue;
+
+    auto fields = StringVectorToIntVector(splits);
+    auto num_fields = fields.size();
+    if (num_fields == 3u) {
+      Arc arc;
+      arc.src_state = fields[0];
+      arc.dest_state = fields[1];
+      arc.label = fields[2];
+
+      if (arc.src_state >= vec.size()) vec.resize(arc.src_state + 1);
```

Comment:
it's better to do `new_size = max(arc.arc_state, arc.dest_state)` and then `resize`, otherwise you may `resize` twice here.

Reply:
Done.